ragna/ui/central_view.py

```diff
@@ -0,0 +1,681 @@
+                                                                    width:fit-content;
+                                                                    grid-column: span 1;
+                                                                    
+                                                                }   
+                                                 
+
+                                                """.replace(
+                                "{{GRID_HEIGHT}}", str(grid_height)
+                            )
+                        ],
+                    ),
+                    # stylesheets=[""" :host {background-color:red; width:100%;} .bk-panel-models-layout-Column { width: 100%; } """ ]
+                ],
+            )
+
+    def on_click_source_info_wrapper(self, event, msg):
+        if self.on_click_chat_info is not None:
+            markdown = "This response was generated using the following data from the uploaded files: <br />\n"
+
+            for i in range(len(msg.msg_data["sources"])):
+                source = msg.msg_data["sources"][i]
+
+                location = ""
+                if source["location"] != "":
+                    location = f": {source['location']}"
+                markdown += (
+                    f"""{(i+1)}. **{source['document']['name']}** {location}\n"""
+                )
+                markdown += "----\n"
+
+            self.on_click_chat_info(
+                event,
+                "Source Info",
+                [
+                    pn.pane.Markdown(
+                        markdown,
+                        dedent=True,
+                        stylesheets=[""" hr { width: 94%; height:1px;  }  """],
+                    ),
+                    # Debug bloc :
+                    # pn.pane.Markdown(
+                    #         f"""Chat ID: {self.current_chat['id']} <br />
+                    #                     Message ID: {msg.msg_data['id']} <br />
+                    #                     Sources: {msg.msg_data['sources']}""",
+                    #         dedent=True,
+                    #     stylesheets=[""" :host {  background-color: red ; }  """]
+                    # ),
+                ],
+            )
+
+    def set_current_chat(self, chat):
+        self.current_chat = chat
+
+    async def chat_callback(
+        self, contents: str, user: str, instance: pn.chat.ChatInterface
+    ):
+        self.current_chat["messages"].append({"role": "user", "content": contents})
+
+        try:
+            answer = self.api_wrapper.answer(self.current_chat["id"], contents)
+
+            self.current_chat["messages"].append(answer["message"])
+
+            yield {
+                "user": "Ragna",
+                "avatar": "🤖",
+                "value": answer["message"]["content"],
+            }
+        except Exception as e:
+            print(e)
+
+            yield {
+                "user": "Ragna",
+                "avatar": "❌",
```

Comment:
1. Are we using `"Ragna"` as user for the panel chat interface? Can we just use the role of the message? Here that would be "system"
2. Can we use our logo for the avatar as we do for the other system messages like the welcome message?